commit b0d73aed19a9f3f242d5a13757016759835c164c
Author: Jason Dillaman <dillaman@redhat.com>
Date:   Mon Oct 14 10:54:44 2019 -0400

    mgr: added 'profile rbd/rbd-read-only' cap
    
    These profiles can be further restricted via 'pool' and
    'namespace' argument optionals.
    
    Signed-off-by: Jason Dillaman <dillaman@redhat.com>

diff --git a/src/mgr/MgrCap.cc b/src/mgr/MgrCap.cc
index 37a256f3fe4..ed5ed46d62c 100644
--- a/src/mgr/MgrCap.cc
+++ b/src/mgr/MgrCap.cc
@@ -154,6 +154,26 @@ void MgrCapGrant::expand_profile() const {
     profile_grants.push_back({{}, {}, {}, "crash post", {}, {}});
     return;
   }
+
+  if (profile == "rbd" || profile == "rbd-read-only") {
+    Arguments filtered_arguments;
+    for (auto& [key, constraint] : arguments) {
+      if (key == "pool" || key == "namespace") {
+        filtered_arguments[key] = std::move(constraint);
+      }
+    }
+
+    mgr_rwxa_t perms = mgr_rwxa_t{MGR_CAP_R};
+    if (profile == "rbd") {
+      perms = mgr_rwxa_t{MGR_CAP_R | MGR_CAP_W};
+    }
+
+    // whitelist all 'rbd_support' commands (restricted by optional
+    // pool/namespace constraints)
+    profile_grants.push_back({{}, "rbd_support", {}, {},
+                              std::move(filtered_arguments), perms});
+    return;
+  }
 }
 
 bool MgrCapGrant::validate_arguments(
